### PR TITLE
client/http: add GetHistoryHotRegions interface and more tests

### PR DESCRIPTION
**Pull request by JmPotato**
### What problem does this PR solve?

Issue Number: ref #7300.

### What is changed and how does it work?

```commit-message
- Add `GetHistoryHotRegions` interface.
- Add more tests.
```

### Check List

Tests

- Unit test
- Integration test

### Release note

```release-note
None.
```

### CODE REVIEW

File: client/http/client.go [hunk 1]
```diff
@@ -191,20 +192,31 @@ func (c *client) execDuration(name string, duration time.Duration) {
 	c.executionDuration.WithLabelValues(name).Observe(duration.Seconds())
 }
 
+// HeaderOption configures the HTTP header.
+type HeaderOption func(header http.Header)
+
+// WithAllowFollowerHandle sets the header field to allow a PD follower to handle this request.
+func WithAllowFollowerHandle(allow bool) HeaderOption {
```
Comment (rleungx) [inline client/http/client.go:199]: How about:
```suggestion
func WithAllowFollowerHandle() HeaderOption {
```

File: client/http/client.go [hunk 2]
```diff
@@ -191,20 +192,31 @@ func (c *client) execDuration(name string, duration time.Duration) {
...
+	return func(header http.Header) {
+		header.Set("PD-Allow-follower-handle", fmt.Sprintf("%t", allow))
```
Comment (CabinfeverB) [inline client/http/client.go:201]: It's better to format it as `PD-Allow-Follower-Handle`. Also, `CanonicalMIMEHeaderKey` has helped us do this, but I think it's clear if we do this.

File: client/http/client.go [hunk 3]
```diff
@@ -191,20 +195,31 @@ func (c *client) execDuration(name string, duration time.Duration) {
 	c.executionDuration.WithLabelValues(name).Observe(duration.Seconds())
 }
 
+// HeaderOption configures the HTTP header.
+type HeaderOption func(header http.Header)
+
+// WithAllowFollowerHandle sets the header field to allow a PD follower to handle this request.
+func WithAllowFollowerHandle() HeaderOption {
+	return func(header http.Header) {
+		header.Set("PD-Allow-Follower-Handle", "true")
```
Comment (HuSharp) [inline client/http/client.go:204]: Do we need to use const for `PD-Allow-Follower-Handle`?

Comment (JmPotato) [reply client/http/client.go:204]: Since there is only one constant right now, I think we can change it later when we introduce more.

**Review by rleungx — APPROVED** [review]

**Review by CabinfeverB — APPROVED** [review]

**Pull request merged**